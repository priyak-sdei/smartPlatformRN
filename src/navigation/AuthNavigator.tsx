import React from 'react';
import { View } from 'react-native';
import { createStackNavigator } from '@react-navigation/stack';
import { Login, Welcome, ForgotPassword, SignUp } from '@pods/auth/index';

export type AuthStackParamList = {
  Auth: undefined;
  Login: undefined;
  Welcome: undefined;
  ForgotPassword: undefined;
  SignUp: undefined;
};

const Stack = createStackNavigator<AuthStackParamList>();

const AuthNavigator = () => (
  <View testID="auth-navigator-root" style={{ flex: 1 }}>
    <Stack.Navigator
      screenOptions={{ headerShown: false }}
      initialRouteName="Welcome"
    >
      <Stack.Screen name="Welcome" component={Welcome} />
      <Stack.Screen name="Login" component={Login} />
      <Stack.Screen name="ForgotPassword" component={ForgotPassword} />
      <Stack.Screen name="SignUp" component={SignUp} />
    </Stack.Navigator>
  </View>
);

export default AuthNavigator;
