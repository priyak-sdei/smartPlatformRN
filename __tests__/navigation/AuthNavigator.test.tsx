import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import AuthNavigator from '../../src/navigation/AuthNavigator';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import renderer, { act } from 'react-test-renderer';

afterEach(cleanup);

describe('AuthNavigator', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <AuthNavigator />
                </NavigationContainer>
            </Provider>
        );
        await waitFor(() => {
            expect(getByTestId('auth-navigator-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(
                <Provider store={store}>
                    <NavigationContainer>
                        <AuthNavigator />
                    </NavigationContainer>
                </Provider>
            );
        });
        expect(tree!.toJSON()).toMatchSnapshot('AuthNavigator component');
    });
}); 