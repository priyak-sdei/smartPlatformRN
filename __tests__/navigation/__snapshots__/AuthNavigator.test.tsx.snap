// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNavigator matches snapshot: AuthNavigator component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="auth-navigator-root"
/>
`;
